lfs.c

```diff
@@ -6318,6 +6316,46 @@ lfs_soff_t lfs_file_size(lfs_t *lfs, lfs_file_t *file) {
     return res;
 }
 
+int lfs_file_movehandle(lfs_t *lfs, lfs_file_t *old_file, 
+        lfs_file_t *new_file) {
+    int err = LFS_LOCK(lfs->cfg);
+    if (err) {
+        return err;
+    }
+    LFS_TRACE("lfs_file_movehandle(%p, %p, %p)",
+        (void*)lfs, (void*)old_file, (void*)new_file);
```

Comment:
If you're making changes, two indentions after first macro/function open paren matches the style in the rest of the codebase:

``` c
LFS_TRACE("lfs_file_movehandle(%p, %p, %p)",
        (void*)lfs, (void*)old_file, (void*)new_file);
```

This avoids confusion with control flow:

```c
if (!lfs_gstate_iszero(&delta)) {
    lfs_gstate_tole32(&delta);
    err = lfs_dir_commitattr(lfs, &commit,
            LFS_MKTAG(LFS_TYPE_MOVESTATE, 0x3ff,
                sizeof(delta)), &delta);
    if (err) {
        if (err == LFS_ERR_CORRUPT) {
            goto relocate;
        }
        return err;
    }
}
```